src/Microsoft.ML.TimeSeries/RootCauseAnalyzer.cs

```diff
@@ -0,0 +1,764 @@
+                }
+
+                double entropy = GetEntropy(dimension.PointDis.Count, dimension.AnomalyDis.Count);
+                if (Double.IsNaN(entropy)) {
+                    entropy = Double.MaxValue;
+                }
+                entropyMap.Add(dimension, entropy);
+
+                double gainRatio = entropy / GetDimensionInstrinsicValue(dimension.PointDis);
+
+                if (Double.IsInfinity(gainRatio)) {
+                    gainRatio = 0;
+                }
+                entropyRatioMap.Add(dimension, gainRatio);
+
+                sumGain += entropy;
+            }
+
+            double meanGain = sumGain / aggDim.Count;
+
+            BestDimension best = FindBestDimension(entropyMap, entropyRatioMap, meanGain, false);
+            return best;
+        }
+
+        private AnomalyDirection GetRootCauseDirection(Point rootCausePoint)
+        {
+            if (rootCausePoint.ExpectedValue < rootCausePoint.Value)
+            {
+                return AnomalyDirection.Up;
+            }
+            else if (rootCausePoint.ExpectedValue > rootCausePoint.Value)
+            {
+                return AnomalyDirection.Down;
+            }
+            else {
+                return AnomalyDirection.Same;
+            }
+        }
+
+        private void GetRootCauseDirectionAndScore(Dictionary<string, Point> dimPointMapping, Dictionary<string, Object> anomalyRoot, RootCause dst, double beta, PointTree pointTree, AggregateType aggType, Object aggSymbol)
+        {
+            Point anomalyPoint = GetPointByDimension(dimPointMapping, anomalyRoot, pointTree, aggType, aggSymbol);
+            if (dst.Items.Count > 1)
+            {
+                //get surprise value and explanary power value
+                List<RootCauseScore> scoreList = new List<RootCauseScore>();
+
+                foreach (RootCauseItem item in dst.Items)
+                {
+                    Point rootCausePoint = GetPointByDimension(dimPointMapping, item.Dimension, pointTree, aggType, aggSymbol);
+                    if (anomalyPoint != null && rootCausePoint != null)
+                    {
+                        Tuple<double, double> scores = GetSupriseAndExplanatoryScore(rootCausePoint, anomalyPoint);
+                        scoreList.Add(new RootCauseScore(scores.Item1, scores.Item2));
+                        item.Direction = GetRootCauseDirection(rootCausePoint);
+                    }
+                }
+
+                //get final score
+                for (int i = 0; i < scoreList.Count; i++)
+                {
+                    if (aggType.Equals(AggregateType.Max) || aggType.Equals(AggregateType.Min))
+                    {
+                        dst.Items[i].Score = 1;
+                    }
+                    else {
+                        dst.Items[i].Score = GetFinalScore(scoreList[i].Surprise, Math.Abs(scoreList[i].ExplainaryScore), beta);
```

Comment:
nit: Please move the { on to the next line

Reply:
> nit: Please move the { on to the next line

updated